string Class   
uint16 centerX
uint16 centerY 
float64 angleX
float64 angleY
float64 distance
